float32 cusum_pos
float32 cusum_angle
float32 x_hat_pos
float32 x_hat_angle
